Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_common.py

```diff
@@ -102,15 +170,250 @@ def draw(self, context):
+
+    bl_idname = "meshsync.install_meshsync"
+    bl_label = "Try to get the unity path from a running server"
+
+    def get_meshsync_path(self):
+        script_file = os.path.realpath(__file__)
+        directory = os.path.dirname(script_file)
+        system = platform.system()
+        if system == 'Windows':
+            return directory+"\\MeshSyncClientBlender\\resources\\com.unity.meshsync"
+        elif system == 'Darwin':
+            return directory+"/MeshSyncClientBlender/resources/com.unity.meshsync"
+        elif system == 'Linux':
+            return directory+"/MeshSyncClientBlender/resources/com.unity.meshsync"
+
+    def add_meshsync_to_unity_manifest(self, path, entry):
+        file = open(path, "r+");
+        data = json.load(file);
+
+        # check if MeshSync is installed on selected project
+        # if not installed, install it
+        found = False
+
+        for package in data['dependencies']:
+            if (package == 'com.unity.meshsync'):
+                found = True
+
+        if(found == False):
+            #install for user
+            dependencies = data["dependencies"];
+            dependencies["com.unity.meshsync"] = entry
+            if (entry == ""):
+                del dependencies["com.unity.meshsync"]
+            file.seek(0)
+            file.truncate(0)
+            json.dump(data, file)
+        
+        file.close()
+
+    def install_meshsync_to_unity_project(self, directory):
+
+        #Modify the Unity Package manifest to add the MeshSync package from disk.
+        if directory[-1] == '/' or directory[-1] == '\\':
+            manifest_path = directory + "Packages/manifest.json"
+        else:
+            manifest_path = directory + "/Packages/manifest.json"
+
+        #TODO replace with release that has the Editor Commands changes
+        manifest_entry = "0.14.5-preview"
+
+        self.add_meshsync_to_unity_manifest(manifest_path, manifest_entry)
+
+    def execute(self, context):
+        directory = context.scene.meshsync_unity_project_path
+        self.install_meshsync_to_unity_project(directory)
+        bpy.ops.meshsync.start_unity('INVOKE_DEFAULT')
+        return {'FINISHED'}
+
+class MESHSYNC_OT_StartUnity(bpy.types.Operator):
+    bl_idname = "meshsync.start_unity"
+    bl_label = "Export Objects"
+
+    def get_editor_version(self, directory):
+        project_version_path = directory + "/ProjectSettings/ProjectVersion.txt"
+        with open(project_version_path, "r+") as file:
+            first_line = file.readline()
+            version = first_line[len("m_EditorVersion: "):]
+            version = version[:-1]
+            return version
+
+    def launch_project(self, editor_path, project_path):
+        os = platform.system()
+        if os == 'Windows':
+            path = editor_path + " -projectPath \"" + project_path + "\""
+        elif os == 'Darwin':
+            path = editor_path + " -projectPath " + project_path
+        elif os == 'Linux':
```

Comment:
The code for Linux and Mac are usually the same, so perhaps we can make the code simpler ?

Reply:
done